pkg/local/sync.go

```diff
@@ -277,7 +277,7 @@ func (s *SyncManager) upload(ctx context.Context, rootPath string, remote *uri.U
 	if err := fileutil.VerifySafeFilename(source); err != nil {
 		return err
 	}
-	dest := filepath.ToSlash(filepath.Join(remote.GetPath(), path))
+	dest := strings.TrimPrefix(filepath.ToSlash(fmt.Sprintf("%s%s%s", remote.GetPath(), uri.PathSeparator, path)), uri.PathSeparator)
```

Comment:
Consider using a method just to keep it clear and consistent
something like ''JoinWithTrailingSlash'

Reply:
It's a nice to have - but I prefer to defer it for now